### PR TITLE
HSEARCH-2073 LuceneHSQuery cannot be deserialized

### PR DESCRIPTION
**Pull request by gustavonalle**
https://hibernate.atlassian.net/browse/HSEARCH-2073

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/query/engine/impl/LuceneHSQuery.java [hunk 1]
```diff
@@ -66,7 +65,7 @@
  * @author Emmanuel Bernard
  * @author Hardy Ferentschik
  */
-public class LuceneHSQuery extends AbstractHSQuery implements HSQuery, Serializable {
```
Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/LuceneHSQuery.java]: why removing the Serializable in this case?

Comment (gustavonalle) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/LuceneHSQuery.java]: It was redundant since I made the parent Serializable

Comment (Sanne) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/LuceneHSQuery.java]: yes I realize it's redundant, but wondering what is the contract you're needing.
In practice, it's `LuceneHSQuery` which you need to be serializable, right?

We could make it serializable without having to mandate the contract on its parent, or we might want to mark both as Serializable as an explicit marker (reminder).

Comment (gustavonalle) [inline engine/src/main/java/org/hibernate/search/query/engine/impl/LuceneHSQuery.java]: > In practice, it's LuceneHSQuery which you need to be serializable, right?
 
Yep, that's the one causing trouble, although I haven't tested yet with ```ElasticSearchHSQueryImpl``` so I though it'd make sense for both to be Serializable

**Comment by Sanne** [comment]
Replaced by #968

**Pull request closed without merging**